packages/components/src/disclosure/index.tsx

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { DisclosureContentProps } from './types';
+
+/**
+ * Accessible Disclosure component that controls visibility of a section of
+ * content. It follows the WAI-ARIA Disclosure Pattern.
+ *
+ * @see https://ariakit.org/reference/disclosure-content
+ *
+ * The plan is to build own API that accounts for future breaking changes
+ * in Reakit (https://github.com/WordPress/gutenberg/pull/28085).
+ */
+const UnforwardedDisclosureContent = (
+	{ visible, children }: DisclosureContentProps,
+	ref: React.ForwardedRef< any >
+) => {
+	const disclosure = Ariakit.useDisclosureStore( { open: visible } );
+
+	return (
+		<Ariakit.DisclosureContent store={ disclosure } ref={ ref }>
```

Comment:
Should we be including rest props here? I'm seeing the CSS class is missing, which was present before:

| Trunk  | This PR |
| ------------- | ------------- |
|  <img width="935" alt="Screenshot 2023-11-02 at 4 29 42 PM" src="https://github.com/WordPress/gutenberg/assets/35543432/41588966-ab5d-4e49-a0a9-7d4f66f25528"> |  <img width="1017" alt="Screenshot 2023-11-02 at 4 28 03 PM" src="https://github.com/WordPress/gutenberg/assets/35543432/0cb55b5d-9604-4d62-b813-4708921e4c16"> |

In this case it doesn't appear to make a difference visually since it seems the CSS class isn't utilized: 
https://github.com/WordPress/gutenberg/blob/a229cc28d3e292a15845f9a8509661a2fa1971c7/packages/edit-widgets/src/blocks/widget-area/edit/index.js#L72-L73

Reply:
Agreed, and there are a few additional props we'll also need to support IMHO, here's a full list: https://ariakit.org/reference/disclosure-content#optional-props

Reply:
Good catch! Fixed via https://github.com/WordPress/gutenberg/pull/55639/commits/7c19debe3d685f1dcbde6f01460c8349be9de400